package/src/bin/omicron-package.rs

```diff
@@ -419,11 +419,15 @@ async fn do_install(
     do_activate(config, install_dir)
 }
 
-fn uninstall_all_omicron_zones() -> Result<()> {
-    zone::Zones::get()?.into_par_iter().try_for_each(|zone| -> Result<()> {
-        zone::Zones::halt_and_remove(zone.name())?;
-        Ok(())
-    })?;
+async fn uninstall_all_omicron_zones() -> Result<()> {
+    const CONCURRENCY_CAP: usize = 32;
+    futures::stream::iter(zone::Zones::get().await?)
+        .map(Ok::<_, anyhow::Error>)
+        .try_for_each_concurrent(CONCURRENCY_CAP, |zone| async move {
+            zone::Zones::halt_and_remove(zone.name()).await?;
+            Ok(())
+        })
+        .await?;
```

Comment:
Rather than using rayon, this is the stream-based equivalent of the same functionality.